src/main/java/bio/terra/service/duos/DuosClient.java

```diff
@@ -46,28 +75,66 @@ String getDatasetUrl(String duosId) {
    * @param duosId DUOS dataset ID
    * @param userRequest authenticated user
    * @return the DuosDataset associated with duosId
-   * @throws DuosDatasetBadRequestException if identifier is malformed
-   * @throws DuosDatasetNotFoundException if no DUOS dataset found
    */
   public DuosDataset getDataset(String duosId, AuthenticatedUserRequest userRequest) {
     HttpHeaders authedHeaders = getHttpHeaders();
     authedHeaders.setBearerAuth(userRequest.getToken());
+    String url = getDatasetUrl(duosId);
+    logger.info("About to GET {} as authenticated user", url);
     try {
       ResponseEntity<DuosDataset> datasetCall =
           restTemplate.exchange(
-              getDatasetUrl(duosId),
-              HttpMethod.GET,
-              new HttpEntity<>(authedHeaders),
-              DuosDataset.class);
+              url, HttpMethod.GET, new HttpEntity<>(authedHeaders), DuosDataset.class);
       return datasetCall.getBody();
-    } catch (HttpClientErrorException ex) {
-      switch (ex.getStatusCode()) {
-        case BAD_REQUEST -> throw new DuosDatasetBadRequestException(
-            "DUOS dataset identifier %s is malformed".formatted(duosId), ex);
-        case NOT_FOUND -> throw new DuosDatasetNotFoundException(
-            "Could not find DUOS dataset for identifier %s".formatted(duosId), ex);
-        default -> throw ex;
-      }
+    } catch (HttpStatusCodeException ex) {
+      throw convertDuosExToDataRepoEx(ex, duosId);
+    }
+  }
+
+  @VisibleForTesting
+  String getApprovedUsersUrl(String duosId) {
+    return getDatasetUrl(duosId) + "/approved/users";
+  }
+
+  /**
+   * This call to DUOS is performed as the TDR SA rather than a supplied authenticated user.
+   *
+   * @param duosId DUOS dataset ID
+   * @return all users who have access to the dataset via an accepted DAR
+   */
+  public DuosDatasetApprovedUsers getApprovedUsers(String duosId) {
+    HttpHeaders saHeaders = getHttpHeaders();
+    saHeaders.setBearerAuth(googleCredentialsService.getApplicationDefaultAccessToken(SCOPES));
+    String url = getApprovedUsersUrl(duosId);
+    logger.info("About to GET {} as TDR SA", url);
+    try {
+      ResponseEntity<DuosDatasetApprovedUsers> approvedUsersCall =
```

Comment:
Totally up to you, but I'd use `var` for each of these `ResponseEntity` types. I don't find that having the type declared makes it clearer, but it's a matter of taste.
```suggestion
      var approvedUsersCall =
```